core/src/test/scala/unit/kafka/log/UnifiedLogTest.scala

```diff
@@ -4188,11 +4188,77 @@ class UnifiedLogTest {
     assertEquals(1, log.logSegments.size)
   }
 
+  @Test
+  def testRetentionOnLocalLogDeletionWhenRemoteCopyDisabled(): Unit = {
+    def createRecords = TestUtils.records(List(new SimpleRecord(mockTime.milliseconds(), "a".getBytes)))
+    val segmentBytes = createRecords.sizeInBytes()
+    val logConfig = LogTestUtils.createLogConfig(segmentBytes = segmentBytes, localRetentionBytes = 1, retentionBytes = segmentBytes * 5,
+          fileDeleteDelayMs = 0, remoteLogStorageEnable = true)
+    val log = createLog(logDir, logConfig, remoteStorageSystemEnable = true)
+
+    // Given 10 segments of 1 message each
+    for (_ <- 0 until 10) {
+      log.appendAsLeader(createRecords, leaderEpoch = 0)
+    }
+    assertEquals(10, log.logSegments.size)
+
+    log.updateHighWatermark(log.logEndOffset)
+    // simulate calls to upload 2 segments to remote storage
+    log.updateHighestOffsetInRemoteStorage(1)
+
+    log.deleteOldSegments()
+    assertEquals(8, log.logSegments.size())
+    assertEquals(0, log.logStartOffset)
+    assertEquals(2, log.localLogStartOffset())
+
+    // add remoteCopyDisabled = true
+    val copyDisabledLogConfig = LogTestUtils.createLogConfig(segmentBytes = segmentBytes, localRetentionBytes = 1, retentionBytes = segmentBytes * 5,
+      fileDeleteDelayMs = 0, remoteLogStorageEnable = true, remoteCopyDisabled = true)
+    log.updateConfig(copyDisabledLogConfig)
+
+    // No local logs will be deleted even though local retention bytes is 1 because there are still logs in remote storage
+    log.deleteOldSegments()
+    assertEquals(8, log.logSegments.size())
+    assertEquals(0, log.logStartOffset)
+    assertEquals(2, log.localLogStartOffset())
+
+    // simulate the remote logs are all deleted due to retention policy
+    log.updateLogStartOffsetFromRemoteTier(2)
+    assertEquals(8, log.logSegments.size())
+    assertEquals(2, log.logStartOffset)
+    assertEquals(2, log.localLogStartOffset())
+
+    // try to delete local logs again, 3 segments will be deleted this time because log start offset == local log start offset,
+    // which means no remote storage is empty. We'll treat this log as local logs and use retention.bytes for retention policy.
```

Comment:
When LSO == LLSO does mean that the remote logs are empty for the partition. Assume that the user configured the remote and local retention values to be same to observe only the segment copy to the remote storage. Then, disabled the remote copy feature. 

In this case, both the remote and local storage has an overlap of all the passive segments. With `hasRemoteLogs` check, we might trigger the local and remote segment retention in parallel.   But, it will still work as we are moving the log-start-offset on deletion of logs from either local/remote.

Reply:
That's a "hard to understand" case :) , but yes, I agree we have to make it clear. I'll update javadoc and tests.

Reply:
@kamalcph , your comment makes me think of 1 situation that we can't handle the local log deletion well.

Suppose we set `local.retention.ms = 1 min`, `retention.ms = 10 min`.
00:00 -> produced and uploaded segment0 to remote storage, and then set `remote.log.copy.disable = true`.
00:01 -> we expect to delete local log segment0, but because `log start offset == local log start offset == 0`, we can't delete local logs, until 10 mins later.

So, if the `retention.bytes = 10TB`, and `local.retention.bytes = 10GB`, that would be very bad.

I was thinking we can add one more check: `(logStartOffset != localLogStartOffset() || localLogStartOffset() < highestOffsetInRemoteStorage())`, in this way, we can know if there are "pending local log segments to be deleted". But if, like you said, the another broker has different segment.baseOffset, say:
 - localLogStartOffset = 9, 
 - highestOffsetInRemoteStorage() = 10
And because the remote copy is disabled, the `highestOffsetInRemoteStorage` won't be increased anymore, and we can't delete local logs, either.

I can't come out any better solution. I'd like to hear your thoughts. @kamalcph @satishd 

If we can't come out a better solution, I'm proposing we `always use "local.retention.ms/bytes" when "remote.log.copy.disable = true"`. I think using `local.retention.ms/bytes` makes sense to me because it is applying to the local logs. WDYT?

Reply:
Can we add one validation to ensure that the remote and local log retention values (time and size) are same when remote copy gets disabled?

With this, both local and remote log deletion become independent of each other and the user will be aware of the data retention settings in the local disk.

Reply:
Thanks @kamalcph !
I was trying to start a new discussion in KIP-950 to change the design (again...), then, when I re-read the KIP, I found the local log retention change was originally defined:

> When tiered storage is disabled or becomes read-only on a topic, the local retention configuration becomes irrelevant, and all data expiration follows the topic-wide retention configuration exclusively.

With that, I think all we need to do is to make the document clear, and follow the original design. I've updated the PR accordingly. Thanks.

Reply:
@showuon Do you plan to add the required local.log and log retention configs for making `remote.log.copy.disable` as true?

Reply:
@satishd , I don't because in KIP-950, we don't mention the validation anywhere. But on second thought, I agree it makes the retention much clear to users. Let me add the validation and mention it in KIP-950 discussion.

Reply:
Updated in https://github.com/apache/kafka/commit/e03e563502d0953e279becce148da5d977bba5cd . Thanks.